src/components/Bullet.tsx

```diff
@@ -442,11 +441,78 @@ const Bullet = ({
     return children.length < Object.keys(thought.childrenMap).length
   })
 
-  // fill =bullet/=style override
+  /** Check if the entire value is wrapped in a single font or span tag. */
+  const isWrappedInSingleTag = (str: string): boolean => {
+    // Check if there's any text before the first tag or after the last tag
+    const firstTagIndex = str.indexOf('<')
+    const lastTagIndex = str.lastIndexOf('>')
+
+    if (firstTagIndex > 0 || lastTagIndex < str.length - 1) {
+      return false
+    }
+
+    const tagStack: string[] = []
+    let currentPos = 0
+
+    while (currentPos < str.length) {
+      // Find next tag
+      const openIndex = str.indexOf('<', currentPos)
+      if (openIndex === -1) break
+
+      const closeIndex = str.indexOf('>', openIndex)
+      if (closeIndex === -1) return false
+
+      const tag = str.substring(openIndex, closeIndex + 1)
+      currentPos = closeIndex + 1
+
+      if (tag.startsWith('</')) {
+        // Closing tag
+        const tagName = tag.slice(2, -1).toLowerCase()
+        // If stack is empty or last opening tag doesn't match, return false
+        if (tagStack.length === 0 || tagStack[tagStack.length - 1] !== tagName) {
+          return false
+        }
+
+        // Remove matching opening tag
+        tagStack.pop()
+
+        // If stack is empty before end of string, only valid if this is the last tag
+        if (tagStack.length === 0 && currentPos <= str.length) {
+          const remainingTags = str.slice(currentPos).match(/<\/?[a-z]+/g)
+          return !remainingTags
+        }
+      } else if (!tag.endsWith('/>')) {
+        // Opening tag (excluding self-closing tags)
+        const tagName = tag.slice(1).split(/[\s>]/)[0].toLowerCase()
+        tagStack.push(tagName)
+      }
+    }
+
+    // Valid if exactly one tag remains in stack (the outer tag)
+    return tagStack.length === 1 && (tagStack[0] === 'font' || tagStack[0] === 'span')
+  }
+
   const fill = useSelector(state => {
-    const bulletId = findDescendant(state, head(simplePath), '=bullet')
-    const styles = getStyle(state, bulletId)
-    return styles?.color
+    const thought = getThoughtById(state, thoughtId)
+    if (!thought) return undefined
+
+    const value = thought.value
+    // Check if the entire value is wrapped in a single font or span tag
+    const isWrappedInTag = isWrappedInSingleTag(value)
+    if (isWrappedInTag) {
+      // Check for background-color in style attribute
+      const bgColorMatch = value.match(/style="[^"]*background-color:\s*([^;"'>]+)[^"]*"/i)
```

Comment:
You can instantiate static regexes at the top of the module for efficiency.

Reply:
I got it.